### PR TITLE
Avoid catching KeyError within custom-scalars

### PR DESCRIPTION
**Pull request by chihuahua**
The custom scalars plugin had been responsible for catching a KeyError
within its `is_active` and `layout_impl` methods. Now, the plugin calls
`PluginRunToTagToContent` to retrieve the runs that have layouts.

Note that `PluginRunToTagToContent` still catches a KeyError within its
implementation. However, the plugin is no longer responsible for
catching the error.

Test Plan: Note that the custom scalars dashboard works when some run
has a layout. The plugin becomes inactive when I remove the events file
containing the layout from the log directory.

### CODE REVIEW

File: tensorboard/plugins/custom_scalar/custom_scalars_plugin.py [hunk 1]
```diff
@@ -286,15 +285,11 @@ def layout_impl(self):
...
+    runs = list(self._multiplexer.PluginRunToTagToContent(metadata.PLUGIN_NAME))
     runs.sort()
     for run in runs:
-      try:
-        tensor_events = self._multiplexer.Tensors(
-            run, metadata.CONFIG_SUMMARY_TAG)
-      except KeyError:
-        # This run lacks a layout.
-        continue
+      tensor_events = self._multiplexer.Tensors(
```
Comment (jart) [inline tensorboard/plugins/custom_scalar/custom_scalars_plugin.py]: Is this loop logic is correct?

Comment (chihuahua) [reply tensorboard/plugins/custom_scalar/custom_scalars_plugin.py]: I think so. The runs retrieved from `PluginRunToTagToContent` are the ones relevant to the plugin.

Comment (jart) [reply tensorboard/plugins/custom_scalar/custom_scalars_plugin.py]: Ah yes, I see. GitHub and Python didn't make it clear to me that the lines that follow were part of the same scope.

Comment (chihuahua) [reply tensorboard/plugins/custom_scalar/custom_scalars_plugin.py]: Ah, I see. Yeah, the block indentation was hidden from the conversations view. As an unrelated aside, I have several quips about GitHub's code review interface.

Comment (jart) [reply tensorboard/plugins/custom_scalar/custom_scalars_plugin.py]: Nothing will ever compare to critique.

**Pull request merged**